src/Neo.VM/ExecutionEngine.cs

```diff
@@ -1448,16 +1447,17 @@ internal protected void ExecuteNext()
                 try
                 {
                     ExecutionContext context = CurrentContext!;
-                    PreExecuteInstruction();
+                    Instruction instruction = context.CurrentInstruction ?? Instruction.RET;
```

Comment:
This will be compiled as
```
if context.CurrentInstruction != null
  Instruction = context.CurrentInstruction
Else
  Instruction = RET
```
Looks good but `context.CurrentInstruction` will be called twice and it makes a lot of things inside that could be cached. I think that it will be faster:
```suggestion
                    Instruction instruction = context.CurrentInstruction;
                    if (instruction == null) instruction = Instruction.RET;
```

Reply:
The expression will be evaluated only once.